### ISSUE TITLE
Can't ssh to centos virtual machine over private network until restart

### ISSUE THREAD

**Issue by x-yuri** [mitchellh/vagrant#7852]
### Vagrant version
Vagrant 1.8.6

### Host operating system
Arch Linux

### Guest operating system
centos/7 box

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "centos/7"
  config.vm.network "private_network", ip: "192.168.33.10"
end
```

### Debug output
https://gist.github.com/x-yuri/121e31d603660b41aadb0f9aa505517d (debug.log and debug2.log)

### Expected behavior
I should be able to log in to virtual machine right after `vagrant up`.

### Actual behavior
I can't log in right away.

### Steps to reproduce
```sh
[yuri@yuri ~/_/1] vagrant destroy -f && vagrant up --debug &> debug.log
==> default: Forcing shutdown of VM...
==> default: Destroying VM and associated drives...
[yuri@yuri ~/_/1] ssh root@192.168.33.10
ssh: connect to host 192.168.33.10 port 22: No route to host
[yuri@yuri ~/_/1] vagrant ssh
[vagrant@localhost ~]$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:c3:c0:db brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 86294sec preferred_lft 86294sec
    inet6 fe80::5054:ff:fec3:c0db/64 scope link
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:f8:0b:53 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::a00:27ff:fef8:b53/64 scope link
       valid_lft forever preferred_lft forever
[vagrant@localhost ~]$ logout
Connection to 127.0.0.1 closed.
[yuri@yuri ~/_/1] vagrant halt && vagrant up --debug &> debug2.log
==> default: Attempting graceful shutdown of VM...
[yuri@yuri ~/_/1] vagrant ssh
Last login: Sat Oct  1 11:46:56 2016 from 10.0.2.2
[vagrant@vlad-vagrant ~]$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:c3:c0:db brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 86341sec preferred_lft 86341sec
    inet6 fe80::5054:ff:fec3:c0db/64 scope link
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:f8:0b:53 brd ff:ff:ff:ff:ff:ff
    inet 192.168.33.10/24 brd 192.168.33.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fef8:b53/64 scope link
       valid_lft forever preferred_lft forever
```

If I'm not mistaken, `eth1` doesn't always get even ipv6 address.

**Comment by chrisroberts** [comment]
Fixed by: #7751

**Comment by chrisroberts** [comment]
Fixed via #7751
